andela/Ah-backend-aquaman
# #164069230 receive email or in-app notifications

#### What does this PR do?
- Enable authenticated user to receive notifications when users they follow publish new articles.
- Enable users receive notifications when articles they favorited have new interactions (like new comments).

#### Description of Task to be completed?
- Add a url for viewing all notifications
- Add a url for viewing unread notifications
- Add a url for viewing single notification
- Add a notification into database
- send email notifications to users

#### How should this be manually tested?
- Clone the [repo](https://github.com/andela/Ah-backend-aquaman), cd into `Ah-backend-aquaman` and checkout the branch `ft-receive-notifications-164069230`
    -  pip install -r requirements.txt
    -  python manage.py migrate
    -  python manage.py runserver
    -  create two accounts and verify your emails
    -  use postman to test the follow endpoints
      - view all notifications: `GET /api/notifications/`
      - view single notification `GET /api/notifications/single/<slug>/`
      - view unread notifications: `GET /api/notifications/unread/`
      - changing notification settings: `POST /api/notification/<type>/`
#### What are the relevant pivotal tracker stories?
- [#164069230](https://www.pivotaltracker.com/story/show/164069230)

#### Screenshots (if appropriate)
*view all notifications
`GET /api/notifications/`

<img width="1035" alt="Screenshot 2019-03-20 at 17 40 02" src="https://user-images.githubusercontent.com/19649761/54693459-be547000-4b37-11e9-9cfa-59c1945fee19.png">



*view unread notifications
`GET /api/notifications/unread/`

<img width="1035" alt="Screenshot 2019-03-20 at 17 40 55" src="https://user-images.githubusercontent.com/19649761/54693627-04113880-4b38-11e9-9826-feaf112a8786.png">



*view single notification
`GET /api/notifications/single/<slug>/`

<img width="1035" alt="Screenshot 2019-03-20 at 17 41 53" src="https://user-images.githubusercontent.com/19649761/54693684-15f2db80-4b38-11e9-8a97-29d59d680652.png">
